File: Editor/src/main/java/org/turbomedia/turboedit/editor/windows/preferences/EditRenderServerWindow.java
package org.turbomedia.turboedit.editor.windows.preferences;

import javafx.scene.Scene;
import javafx.stage.Modality;
import javafx.stage.Stage;
import org.turbomedia.turboedit.editor.panes.preferences.EditRenderServerPane;
import org.turbomedia.turboedit.editor.renderer.RenderServerEntry;

import static org.turbomedia.turboedit.editor.Editor.ICON;
import static org.turbomedia.turboedit.editor.Editor.TITLE;
import static org.turbomedia.turboedit.editor.misc.Locale.GetText;

public class EditRenderServerWindow extends Stage {
    public EditRenderServerWindow(RenderServerEntry entry) {
        setTitle(TITLE + " - " + GetText("title.edit_render_server"));
        getIcons().add(ICON);
        setWidth(500);

        setScene(new Scene(new EditRenderServerPane(this, entry)));

        setResizable(false);
        centerOnScreen();
        initModality(Modality.APPLICATION_MODAL);
        showAndWait();
    }
}

File: Editor/src/main/kotlin/turboedit/editor/windows/preferences/EditRenderServerWindow.kt
package turboedit.editor.windows.preferences

import javafx.scene.Scene
import javafx.stage.Modality
import javafx.stage.Stage
import turboedit.editor.Editor
import turboedit.editor.misc.Locale
import turboedit.editor.panes.preferences.EditRenderServerPane
import turboedit.editor.renderer.RenderServerEntry

class EditRenderServerWindow(entry: RenderServerEntry, responseID: String?) : Stage() {
    init {
        title = Editor.Companion.TITLE + " - " + Locale.getText("title.edit_render_server")
        icons.add(Editor.Companion.ICON)
        width = 500.0;
        isResizable = false
        scene = Scene(EditRenderServerPane(this, entry, responseID));

        centerOnScreen()
        initModality(Modality.APPLICATION_MODAL)
        showAndWait()
    }
}
